config/block_configuration.go

```diff
@@ -159,7 +159,7 @@ func (b *BlockConfiguration) ToBlock(orgName string, stackId int64) types.Block
 }
 
 func (b *BlockConfiguration) ApplyChangesTo(ic core.IacContext, updater core.WorkspaceConfigUpdater) error {
-	updater.UpdateSchema(b.ModuleSource, b.ModuleVersion)
+	updater.UpdateSchema(b.ModuleSource, b.ModuleConstraint, b.ModuleVersion)
```

Comment:
This is probably fine but do you mind giving me some insight into what this is and why its added here?

Reply:
The purpose of `ApplyChangesTo` is to take the IaC definition and apply it on a WorkspaceConfig.
`ModuleConstraint` *now* refers to `module_version` in the yaml.
The `ModuleVersion` field is the effective result of resolving the module constraint.
We want to preserve both in the WorkspaceConfig and this change ensures that.